src/Client.ts

```diff
@@ -109,13 +110,11 @@ class Client {
      * @memberof Client
      */
     public async login(email: string, password: string) {
-        if (typeof email === "undefined" || email === "") {
-            throw new InvalidArgumentError("Email is required.");
+        if (!email || !password) {
+            throw new InvalidArgumentError("Email and password are required.");
         }
 
-        if (typeof password === "undefined" || password === "") {
-            throw new InvalidArgumentError("Password is required");
-        }
+        if (!validate(email)) throw new InvalidArgumentError("Invalid email");
```

Comment:
Do we really need to do email validation here? I think we should simply let the server return the error in the event they use an invalid email to reduce the amount of dependencies down.

Reply:
Well, if we are already doing input validation, why not validate the email without making an actual request? This way we know exactly what went wrong and can tell that to the user as early as possible. The package used is also well-known and lightweight - it's not like there are too many dependencies used as well.

Reply:
I was just basically doing very simple validation, make sure it exists/isnt undefined in the case of a variable/config issue or what not. I'd say this is fine, but just seems a little excessive.

Reply:
hm... My reasoning was that the server could err in unpredicted ways, right? It is not a guarantee (ok, knowing how SE treats chat it can be a safe bet nothing changes) that the request fails or fails in the same way, thus we cannot reliably tell the user what went wrong, only that their message failed. And here we save one network round trip + tell the user exactly what's wrong for the price of one small package. Although I think the check can be trimmed further because I still left the `!email` check but, actually, I think it is not needed - as validation will fail on an empty email.